bin/start.sh

```
@@ -127,13 +150,22 @@ preflight_check() {
     tail -5 "$npm_log"
     log_success "Dependencies installed"
 
-    # Step 2: Install git hooks (pre-push validation)
+  fi
+
+  # First-time git hooks (only when node_modules was just created).
+  if [ ! -d "$ROOT_DIR/.git/hooks" ] || [ ! -f "$ROOT_DIR/.git/hooks/pre-push" ]; then
     if [ -f "$SCRIPT_DIR/install-git-hooks.sh" ]; then
       bash "$SCRIPT_DIR/install-git-hooks.sh" 2>/dev/null && \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)